Lib/rlcompleter.py

```diff
@@ -176,6 +176,16 @@ def attr_matches(self, text):
                 if (word[:n] == attr and
                     not (noprefix and word[:n+1] == noprefix)):
                     match = "%s.%s" % (expr, word)
+                    if isinstance(getattr(type(thisobject), word, None),
+                                  property):
+                        # bpo-44752: thisobject.word is a method decorated by
+                        # `@property`. What follows applies a postfix if
+                        # thisobject.word is callable, but know we know that
+                        # this is not callable (because it is a property).
+                        # Also, getattr(thisobject, word) will evaluate the
+                        # property method, which is not desirable.
+                        matches.append(match)
+                        continue
                     try:
```

Comment:
This is a code style issue not related to the bpo, but I don't see why we need the try/except here at all. Can it be simplified and cleaned up?

```python
if value := getattr(thisobject, word, None):
    matches.append(self._callable_postfix(value, match))
else:
    matches.append(match)
```

Reply:
For reference, this is the current code:

```python
try:
    val = getattr(thisobject, word)
except Exception:
    pass  # Include even if attribute not set
else:
    match = self._callable_postfix(val, match)
matches.append(match)

```

Reply:
The equivalence between old code and new code depends on the fact that `_callable_postfix` simply returns an untouched `match` if `val` isn't callable (which is true if it's `None`). Moreover, while I don't know of any standard callables which are falsy, it's relatively easy to come up with one:
```pycon
>>> class falsycallable:
...   def __call__(self):
...     print("called it!")
...   def __bool__(self):
...     return False
...
>>> fc = falsycallable()
>>> fc()
called it!
>>> if fc: print("True!")
...
>>>
```

So if you'd like to refactor this, which I'd do in a separate PR (although for simplicity you can use the same BPO number), then I suggest checking for `is not None` explicitly.

Reply:
Ok, thank you for the feedback – that's a great point about the possibility of a falsy callable. See https://github.com/python/cpython/pull/27433